**My Inventory**

In [71]:
record= {100:{"name":"Perk","price":5,"quantity":20,"weight":20,'isAvailable':1,"manufacture_date":"Best Before 3 months","expiry":"2021/11/02","type":"chocolate"},
      101:{"name":"Kitkat","price":5,"quantity":20,"weight":10,"manufacture_date":"Best Before 3 months","expiry":"2021/11/03","type":"chocolate"},
      102:{"name":"5star","price":5,"quantity":20,"weight":15,"manufacture_date":"Best Before 3 months","expiry":"2021/11/04","type":"chocolate"},
      103:{"name":"Cadbury","price":5,"quantity":20,"weight":20,"manufacture_date":"Best Before 3 months","expiry":"2021/11/02","type":"chocolate"},
      104:{"name":"munch","price":10,"quantity":20,"weight":10,"manufacture_date":"Best Before 3 months","expiry":"2021/11/02","type":"chocolate"},
      105:{"name":"Mango-Juice","price":10,"quantity":20,"weight":20,"manufacture_date":"Best Before 4 months","expiry":"2021/11/05","type":"drinks"},
      106:{"name":"Fruti","price":20,"quantity":20,"weight":50,"manufacture_date":"Best Before 4 months","expiry":"2021/11/06","type":"drinks"},
      107:{"name":"Fizz","price":30,"quantity":20,"weight":50,"manufacture_date":"Best Before 4 months","expiry":"2021/11/06","type":"drinks"},
      108:{"name":"Nacho","price":50,"quantity":20,"weight":20,"manufacture_date":"Best Before 5 months","expiry":"2021/11/07","type":"snacks"},
      109:{"name":"Sting","price":5,"quantity":20,"weight":20,"manufacture_date":"Best Before 4 months","expiry":"2021/11/06","type":"drinks"},
      110:{"name":"bingo","price":5,"quantity":20,"weight":20,"manufacture_date":"Best Before 5 months","expiry":"2021/11/07","type":"snacks"},
      111:{"name":"fruti","price":5,"quantity":20,"weight":10,"manufacture_date":"Best Before 4 months","expiry":"2021/11/06","type":"drinks"},
      112:{"name":"Mountain Dew","price":25,"quantity":20,"weight":20,"manufacture_date":"Best Before 4 months","expiry":"2021/11/06","type":"drinks"},
      113:{"name":"Oreo","price":15,"quantity":20,"weight":20,"manufacture_date":"Best Before 6 months","expiry":"2021/11/16","type":"biscuits"},
      114:{"name":"Cocacola","price":20,"quantity":20,"weight":200,"manufacture_date":"Best Before 4 months","expiry":"2021/11/16","type":"drinks"},
      115:{"name":"Parle-G","price":5,"quantity":20,"weight":20,"manufacture_date":"Best Before 6 months","expiry":"2021/11/06","type":"biscuits"},
      116:{"name":"mari-gold","price":5,"quantity":20,"weight":50,"manufacture_date":"Best Before 6 months","expiry":"2021/11/20","type":"biscuits"},
      117:{"name":"Dark Chocolate","price":30,"quantity":20,"weight":20,"manufacture_date":"Best Before 6 months","expiry":"2021/11/16","type":"chocolate"},
      118:{"name":"Amul choco Drink","price":100,"quantity":20,"weight":100,"manufacture_date":"Best Before 4 months","expiry":"2021/11/20","type":"drinks"},
      119:{"name":"Mirinda","price":12,"quantity":20,"weight":20,"manufacture_date":"Best Before 4 months","expiry":"2021/11/26","type":"drinks"},
      120:{"name":"Maggi Noodles","price":18,"quantity":20,"weight":20,"manufacture_date":"Best Before 7 months","expiry":"2021/11/27","type":"noodles"},
      121:{"name":"Lays","price":10,"quantity":20,"weight":20,"manufacture_date":"Best Before 5 months","expiry":"2021/11/27","type":"snacks"},
      122:{"name":"Yipee Noodles","price":25,"quantity":20,"weight":25,"manufacture_date":"Best Before 7 months","expiry":"2021/11/17","type":"noodles"},
      123:{"name":"Chatpat Wheels","price":5,"quantity":20,"weight":30,"manufacture_date":"Best Before 5 months","expiry":"2021/11/07","type":"snacks"},
      124:{"name":"Peanuts","price":500,"quantity":20,"weight":250,"manufacture_date":"Best Before 5 months","expiry":"2021/11/30","type":"nuts"},
      125:{"name":"Sneakers","price":10,"quantity":20,"weight":10,"manufacture_date":"Best Before 6 months","expiry":"2021/11/06","type":"chocolate"},
      126:{"name":"Dairy Milk silk","price":100,"quantity":20,"weight":50,"manufacture_date":"Best Before 6 months","expiry":"2021/11/26","type":"chocolate"},
      127:{"name":"Milkybar","price":5,"quantity":20,"weight":10,"manufacture_date":"Best Before 6 months","expiry":"2021/11/17","type":"chocolate"},
      128:{"name":"Lassi","price":20,"quantity":20,"weight":300,"manufacture_date":"Best Before 4 months","expiry":"2021/11/16","type":"drinks"},
      129:{"name":"Red Bull","price":120,"quantity":20,"weight":200,"manufacture_date":"Best Before 4 months","expiry":"2021/11/16","type":"drinks"},
      130:{"name":"Engage","Quantity":20,"Price":189,"manufacture_date":"Best before 9 months","expiry":"2022/10/23","type":"perfume"},
      131:{"name":"Acqua Di Gio","quantity":20,"price":5500,"weight":900,"manufacturer_date":"Best before 8 months","expiry":"2022/12/13","type":"perfume"},
      132:{"name":"Jean Paul Gaultier","quantity":10,"price":5000,"weight":850,"manufacturer_date":"Best before 8 months","expiry":"2022/12/13","type":"perfume"},
      133:{"name":"Badam(400gm)","quantity":240,"price":409,"weight":400,"manufacturer_date":"Best before 10 months","expiry":"2021/11/23","type":"dry fruits"},
      134:{"name":"Khajoor(400gm)","quantity":240,"price":409,"weight":400,"manufacturer_date":"Best before 10 months","expiry":"2021/11/23","type":"dry fruits"},
      135:{"name":"Dates(400gm)","quantity":240,"price":409,"weight":400,"manufacturer_date":"Best before 10 months","expiry":"2021/11/23","type":"dry fruits"},
      126:{"name":"Pista(200gm)","quantity":240,"Price":400,"weight":200,"manufacturer_date":"Best before 10 months","expiry":"2021/11/25","type":"dry fruits"},
      137:{"name":"Almonds(345gm)","quantity":240,"price":700,"weight":345,"manufacturer_date":"Best before 10 months","expiry":"2021/11/27","type":"dry fruits"},
      138:{"name":"Sprit","quantity":50,"price":40,"weight":1.5,"manufacturer_date":"Best before 7 months","expiry":"2021/11/30","type":"drinks"},
      139:{"name":"Pepsi","quantity":50,"price":40,"weight":2,"manufacturer_date":"Best before 4 months","expiry":"2021/12/01","type":"drinks"},
      }

**Creating product.json and dumping all the record into it. 
Adding records into product file**

In [72]:
import json
js=json.dumps(record)
x=open("product.json",'w')
x.write(js)
x.close()
x=open("product.json",'r')
txt=x.read()
x.close()
record2=json.loads(txt)


In [73]:
# Displaying the Type Of Items in my Cart

print("Welcome to Prabhas Inventory")
print("======= == ======= =========")
print()
print("We have")
print("1.Snacks\n2.Drinks\n3.Chocolates\n4.Noodles\n5.Nuts\n6.Mens Perfume\n7.Womens Perfume\n8.Dry Fruits\n9.Biscuits")
print()

Welcome to Prabhas Inventory
======= == ======= =========

We have
1.Snacks
2.Drinks
3.Chocolates
4.Noodles
5.Nuts
6.Mens Perfume
7.Womens Perfume
8.Dry Fruits
9.Biscuits



**1.Generating Bill for the customer purchase and creating a file for Sales**

**2.Updating product file after product is sold**

In [74]:
import time
#getting products info from product file !!
x=open("product.json",'r')
txt=x.read()
x.close()
list2=json.loads(txt)


#Creating bill for product purchase !!
n=input("Enter the Product id          :   ")
m=int(input("Enter the Quantity        :   "))
user = input("Enter Customer Name      :   ")
phone = input("Enter Phone Number      :   ")

if n in record2:
  product =record2[n]
  if m<=product["quantity"]:

        print()
        print("Adipurush Inventory Store")
        print("========= ========= =====")
        print()
        print("Customer Name             :",user)
        print("Phone Number              :",phone)
        print("Product Name              :",product["name"])
        print("Price per piece in rupees :",product['price'])
        print("***********************************************")
        print("Bill amount in rupees is  :",product['price']*m)
        print("Weight of products        :",product['weight']*m,"mg")
        print()
        current_time=time.ctime()
        print(current_time)
        print("***********************************************")
        print("Thank you for shopping! visit again ")
        #product_remain=product['quantity']-m

  else:
      print("sorry for inconvenience but we have limited stock")
      m=0
else:
  print("product not found")
  m=0
#remaining quantity of product 
product_remain=product['quantity']-m
record2[n]["quantity"]=product_remain

#adding sold products info into sold file
sold_list={n:{'Product Name':product["name"],'Quantity Sold':m,'Amount':product['price']*m}}
se=json.dumps(sold_list)
x=open("sales.json",'a')
x.write(se)
x.close()
#updating product file after product is sold
js=json.dumps(record2)
x=open("product.json",'w')
x.write(js)
x.close()

Enter the Product id          :   102
Enter the Quantity        :   10
Enter Customer Name      :   Prabhas
Enter Phone Number      :   123456789

Adipurush Inventory Store
========= ========= =====

Customer Name             : Prabhas
Phone Number              : 123456789
Product Name              : 5star
Price per piece in rupees : 5
***********************************************
Bill amount in rupees is  : 50
Weight of products        : 150 mg

Mon Sep  6 13:40:12 2021
***********************************************
Thank you for shopping! visit again 


**Updating Existing Products and also updating in product file,also adding new Products into the inventory**

In [75]:
x=open("product.json",'r')
txt=x.read()
x.close()
record3=json.loads(txt)

#updating products in record

n=input("Enter the Product id      :")
if n in record3:
  m=int(input("Enter the Quantity        :"))
  k=record3[n]['quantity']+m
  record3[n]['quantity']=k

else:
  p=input("Enter Product Name           :  ")
  m=int(input("Enter the Quantity       :  "))
  o=int(input("Enter Price              :  "))
  w=float(input("Enter weight           :  "))
  m=input("Enter Manufacturer Date      :  ")
  e=input("Enter Expiry date            :  ")
  t=input("Enter Type                   :  ")
  
  record3[n]={'name':p,'price per piece':o,'quantity':m,'weight':w,'manufacturer_date':m,'expiry':e,'type':t}

#updating products in product file

ne=json.dumps(record3)
x=open("product.json",'w')
x.write(ne)
x.close()
print("product added ") #Product Added into Inventory


Enter the Product id      :140
Enter Product Name           :  Pen
Enter the Quantity       :  10
Enter Price              :  10
Enter weight           :  10
Enter Manufacturer Date      :  Best Before 10 days
Enter Expiry date            :  2021/12/12
Enter Type                   :  Stationary
product added 


**The below code shows the updated product quantity and also shows if new product is stored in the inventory**

In [76]:
x=open("product.json",'r')
txt=x.read()
x.close()
print(txt) 

{"100": {"name": "Perk", "price": 5, "quantity": 20, "weight": 20, "isAvailable": 1, "manufacture_date": "Best Before 3 months", "expiry": "2021/11/02", "type": "chocolate"}, "101": {"name": "Kitkat", "price": 5, "quantity": 20, "weight": 10, "manufacture_date": "Best Before 3 months", "expiry": "2021/11/03", "type": "chocolate"}, "102": {"name": "5star", "price": 5, "quantity": 10, "weight": 15, "manufacture_date": "Best Before 3 months", "expiry": "2021/11/04", "type": "chocolate"}, "103": {"name": "Cadbury", "price": 5, "quantity": 20, "weight": 20, "manufacture_date": "Best Before 3 months", "expiry": "2021/11/02", "type": "chocolate"}, "104": {"name": "munch", "price": 10, "quantity": 20, "weight": 10, "manufacture_date": "Best Before 3 months", "expiry": "2021/11/02", "type": "chocolate"}, "105": {"name": "Mango-Juice", "price": 10, "quantity": 20, "weight": 20, "manufacture_date": "Best Before 4 months", "expiry": "2021/11/05", "type": "drinks"}, "106": {"name": "Fruti", "price"

**Below code gives the sales record . The Output gives all the sales**

In [77]:
s=open("sales.json","r")
fs=s.read()
s.close()
print(fs)

{"120": {"product name": "Maggi Noodles", "quantity sold": 5, "amount": 90}}{"107": {"product name": "Fizz", "quantity sold": 5, "amount": 150}}{"107": {"product name": "Fizz", "quantity sold": 10, "amount": 300}}{"105": {"product name": "Mango-Juice", "quantity sold": 10, "amount": 100}}{"107": {"Product Name": "Fizz", "Quantity Sold": 10, "Amount": 300}}{"102": {"Product Name": "5star", "Quantity Sold": 10, "Amount": 50}}


In [78]:
import time
print()
current_time=time.ctime()
print(current_time)


Mon Sep  6 13:42:57 2021


**1)In the above Code, I have added a new Product(PEN) whose Product_ID is "140"and this product is updated in my inventory also.**
**2)Customer have purchased product(5star) whose Product_ID is "102" and this sales product is added into my sales.json file.**